src/pool/disjoint/pool_disjoint_heap_manager.cpp

```diff
@@ -0,0 +1,112 @@
+// Copyright (C) 2023 Intel Corporation
+// Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#include <umf/memory_pool_ops.h>
+#include <umf/memory_provider.h>
+
+#include "pool/pool_disjoint.h"
+#include "pool_disjoint.hpp"
+
+struct disjoint_memory_pool {
+    umf_memory_provider_handle_t mem_provider;
+    usm::DisjointPool *disjoint_pool;
+};
+
+enum umf_result_t
+disjoint_pool_initialize(umf_memory_provider_handle_t *providers,
+                         size_t numProviders, void *params, void **pool) {
+    struct umf_disjoint_pool_params *pub_params = (struct umf_disjoint_pool_params *)params;
+    usm::DisjointPoolConfig config{};
+    config.SlabMinSize = pub_params->SlabMinSize;
+    config.MaxPoolableSize = pub_params->MaxPoolableSize;
+    config.Capacity = pub_params->Capacity;
+    config.MinBucketSize = pub_params->MinBucketSize;
+    config.CurPoolSize = pub_params->CurPoolSize;
+    config.PoolTrace = pub_params->PoolTrace;
+
+    struct disjoint_memory_pool *pool_data =
+        (struct disjoint_memory_pool *)malloc(
+            sizeof(struct disjoint_memory_pool));
+    if (!pool_data) {
+        fprintf(stderr, "cannot allocate memory for metadata\n");
+        abort();
+    }
+
+    pool_data->mem_provider = providers[0];
+    pool_data->disjoint_pool = new usm::DisjointPool();
+    pool_data->disjoint_pool->initialize(providers, numProviders, config);
+
+    *pool = (void *)pool_data;
+
+    (void)params;
```

Comment:
nit: is this needed?

Reply:
Removed: https://github.com/oneapi-src/unified-memory-framework/pull/17